src/components/navigation/left/ServerListSidebar.tsx

```diff
@@ -251,6 +259,44 @@ export default observer(() => {
     const homeActive =
         typeof server === "undefined" && !path.startsWith("/invite");
 
+    // visually selected server id and channel id
+    let [selectedServer, setSelectedServer] = useState<string | null>(null);
+
+    const openServer = useDebounceCallback(
+        (id: string) => {
+            history.push(state.layout.getServerPath(id));
+            setSelectedServer(null);
+        },
+        [],
+        275,
+    );
+
+    // keybinds
+    // TODO: visually select channel, server, etc.. and *then* change the state, this will reduce lag.
```

Comment:
There's a cheat function in `lib/defer.ts` called `Defer` which takes a function and sets a timeout in 0ms, this would make it run after it is done rendering.

Reply:
This is actually a TODO I forgot to remove, `openServer` is what implements what I was describing here.